File: source/input.cpp
#include "../../Include/PowerRonin/Input.hpp"
#include "gui/gui_headers.hpp"

namespace PowerRonin
{
	auto Input::IsKeyDown(const Key key) const noexcept -> bool
	{
		return ImGui::GetIO().KeysDown[static_cast<std::size_t>(key)];
	}

	auto Input::IsMouseButtonDown(const MouseButton mouseButton) const noexcept -> bool
	{
		return ImGui::IsMouseDown(static_cast<ImGuiMouseButton>(mouseButton));
	}

	auto Input::MousePosition() const noexcept -> Vector2<>
	{
		const auto position = ImGui::GetMousePos();
		return {position.x, position.y};
	}
}

File: source/managed/power_ronin/Src/Core/Input.cs
using System.Runtime.CompilerServices;
using PowerRonin.Mathematics;

namespace PowerRonin.Core
{
    /// <summary>
    ///     Contains methods for input fetching.
    /// </summary>
    public static class Input
    {
        private static Vector2 _mousePos = Vector2.Zero;

        /// <summary>
        ///     The mouse cursor position.
        /// </summary>
        public static Vector2 MousePosition => _mousePos;

        /// <summary>
        /// </summary>
        /// <param name="key">The key.</param>
        /// <returns>True if the key is currently pressed, else false.</returns>
        public static bool IsKeyDown(Key key)
        {
            return NativeRuntime.InputIsKeyDown(key);
        }

        /// <summary>
        /// </summary>
        /// <param name="mouseButton">The mouse button.</param>
        /// <returns>True if the mouse button is currently pressed, else false.</returns>
        [MethodImpl(MethodImplOptions.InternalCall)]
        public static bool IsMouseDown(MouseButton mouseButton)
        {
            return NativeRuntime.InputIsMouseDown(mouseButton);
        }

        internal static void Update()
        {
            NativeRuntime.InputGetCursorPosition(out var x, out var y);
            _mousePos.X = x;
            _mousePos.Y = y;
        }
    }
}